File: Objc/Recursive/GCD/GCD.m
//
//  GCD.m
//  Algorithm
//
//  Created by Kuan-Wei Lin on 3/8/17.
//  Copyright © 2017 TaiwanMobile. All rights reserved.
//

#import "GCD.h"

@implementation GCD

- (int)GCD:(int)a b:(int)b
{
    if ((a % b) == 0)
    {
        return b;
    }
    else
    {
        return [self GCD:b b:(a%b)];
    }
}

@end

File: Swift/Recursive/GCD.swift
//
//  GCD.swift
//  Algorithm_Swift
//
//  Created by Kuan-Wei Lin on 3/16/17.
//  Copyright © 2017 cracktheterm. All rights reserved.
//

import UIKit

class GCD: NSObject
{
    func GCD(a:Int, b:Int) -> Int
    {
        if a % b == 0
        {
            return b;
        }
        else
        {
            return GCD(a: b, b: a%b);
        }
    }
}
